libs/coin-framework/src/api/types.ts

```diff
@@ -5,18 +5,28 @@ export type BlockInfo = {
 };
 
 export type Operation = {
-  hash: string;
-  address: string;
   type: string;
-  value: bigint;
-  fee: bigint;
-  block: BlockInfo;
+  // This operation corresponds to the index-th event triggered bu the original transaction
+  operationIndex: number;
   senders: string[];
   recipients: string[];
-  date: Date;
-  transactionSequenceNumber: number;
+  value: bigint;
+  // Asset is not defined when dealing with native currency
+  asset?: AssetInfo;
   // Field containing dedicated value for each blockchain
   details?: Record<string, unknown>;
+  tx: {
+    // One tx can trigger multiple operations, hence multiple operations with the same hash
+    hash: string;
+    // In which block this operation's related tx was included
+    block: BlockInfo;
+    fees: bigint;
```

Comment:
I think we also need `fees_payer` here. There are some ambiguous cases where we can't know for sure if fees should be ignored/subtracted when computing balances from operations, eg when using delegation (not particularly tron, but in general)

Reply:
TODO in another PR

Reply:
ok!